intermediate_source/parametrizations.py

```diff
@@ -0,0 +1,353 @@
+# -*- coding: utf-8 -*-
+"""
+Parametrizations Tutorial
+=========================
+**Author**: `Mario Lezcano <https://github.com/lezcano>`_
+
+Regularizing deep-learning models is a surprisingly challenging task.
+Classical techniques such as penalty methods often fall short when applied
+on deep models due to the complexity of the function being optimized.
+This is particularly problematic when working with ill-conditioned models.
+Examples of these are RNNs trained on long sequences and GANs. A number
+of techniques have been proposed in the recent years to regularize these
+models and improve their convergence. On recurrent models, it has been
+proposed to control the singular values of the recurrent kernel for the
+RNN to be well-conditioned. This can be achieved, for example, by making
+the recurrent kernel `orthogonal <https://en.wikipedia.org/wiki/Orthogonal_matrix>`_.
+Another way to regularize recurrent models is via
+"`weight normalization <https://pytorch.org/docs/stable/generated/torch.nn.utils.weight_norm.html>`_".
```

Comment:
I put them in quotation marks to signal that they are a proper name referring to a particular method. Otherwise "weight normalisation" could mean any sort of normalisation that you put on the weights.